Add available for hire button

padulam opened the pull request:
<!-- freeCodeCamp Pull Request Template -->

<!-- IMPORTANT Please review https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/CONTRIBUTING.md for detailed contributing guidelines -->
<!-- Help with PRs can be found at https://gitter.im/FreeCodeCamp/Contributors -->
<!-- Make sure that your PR is not a duplicate -->

#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
<!-- What type of change does your code introduce? After creating the PR, tick the checkboxes that apply. -->
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask in the Help Contributors room linked above. We're here to help! -->
- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #15357

#### Description
<!-- Describe your changes in detail -->
This update adds an available for hire button to the user settings page. The default setting for this button is 'unavailable'. The user can toggle the button from available/unavailable, similar to the other buttons on the settings page.

padulam committed:
Add available for hire button

raisedadead commented on server/services/user.js:
```diff
@@ -16,6 +16,7 @@ const publicUserProps = [
   'isGithubCool',
 
   'isLocked',
+  'isAvailable',
```
This key should be better named to `isAvailableForHire`

raisedadead commented on server/boot/user.js:
```diff
@@ -463,6 +463,7 @@ module.exports = function(app) {
           isGithubCool: true,
           isCheater: true,
           isLocked: true,
+          isAvailable: true,
```
This key should be better named to `isAvailableForHire`

raisedadead commented on server/boot/settings.js:
```diff
@@ -82,6 +82,11 @@ export default function settingsController(app) {
   api.post(
+    '/toggle-available-for-hire',
+    ifNoUser401,
+    toggleUserFlag('isAvailable')
```
This key should be better named to `isAvailableForHire`

raisedadead commented on common/models/user.json:
```diff
@@ -120,6 +120,11 @@
       "description": "Campers profile does not show challenges/certificates to the public",
       "default": false
     },
+    "isAvailable": {
```
This key should be better named to `isAvailableForHire`

raisedadead commented on common/app/routes/settings/redux/update-user-saga.js:
```diff
@@ -17,6 +17,7 @@ import combineSagas from '../../../../utils/combine-sagas';
 
 const urlMap = {
   isLocked: 'lockdown',
+  isAvailable: 'available-for-hire',
```
This key should be better named to `isAvailableForHire`

raisedadead commented on common/app/routes/settings/components/Settings.jsx:
```diff
@@ -219,6 +227,22 @@ export class Settings extends React.Component {
+            xs={ 12 }
+            >
+            <JobSettings
+              isAvailable={ isAvailable }
```
This key should be better named to `isAvailableForHire`

raisedadead commented on common/app/routes/settings/components/Job-Settings.jsx:
```diff
@@ -0,0 +1,38 @@
+          className={ className }
+          onClick={ toggle }
+          >
+          { isAvailable ? 'Available' : 'Unavailable' }
```
This key should be better named to `isAvailableForHire`

raisedadead commented on common/app/routes/settings/components/Job-Settings.jsx:
```diff
@@ -0,0 +1,38 @@
+import classnames from 'classnames';
+
+const propTypes = {
+  isAvailable: PropTypes.bool,
```
This key should be better named to `isAvailableForHire`

raisedadead reviewed (changes requested):
This key should be better named to `isAvailableForHire` for sake of understanding

padulam commented:
@raisedadead I updated all instances of the isAvailable key to isAvailableForHire. I also updated the toggleIsAvailable function to toggleIsAvailableForHire. Thanks!

raisedadead commented:
Awesome! Changes LGTM, I'll let @BerkeleyTrue merge this.

Thanks a lot for this pull request and great work @padulam!!
